shared-module/src/utils/urlManipulation.ts

```diff
@@ -0,0 +1,9 @@
+export function updateQueryStringParameter(url: string, key: string, value: string) {
+  const parsedUrl = new URL(url)
+  if (parsedUrl.searchParams.has(key)) {
+    parsedUrl.searchParams.delete(key)
+  }
+  // Need to encode, because otherwise spaces would be unencoded + signs
```

Comment:
The code here does work but the encoding is unnecesary. See:
```
~ $  node
Welcome to Node.js v17.7.2.
Type ".help" for more information.
> const url = "https://example.com/something?foo=bar&bar=foo"
undefined
> const url2 = "https://example.com/wat?height=300"
undefined
> const parsedUrl = new URL(url2)
undefined
> parsedUrl.searchParams.append("url", url)
undefined
> parsedUrl.toString()
'https://example.com/wat?height=300&url=https%3A%2F%2Fexample.com%2Fsomething%3Ffoo%3Dbar%26bar%3Dfoo'
```

Reply:
If title has spaces in it and you append them without encoding, it will turn the spaces to + signs rather than %20

Reply:
I see